### PR TITLE
New template divisions nested-form

### PR DESCRIPTION
**Pull request by jenachoi**
User can update label, starting page, and ending page of template divisions from the exam template.

### CODE REVIEW

File: app/models/exam_template.rb [hunk 1]
```diff
@@ -12,6 +12,7 @@ class ExamTemplate < ActiveRecord::Base
                                         only_integer: true }
 
   has_many :template_divisions, dependent: :destroy
+  accepts_nested_attributes_for :template_divisions, allow_destroy: true, :update_only => true
```
Comment (david-yz-liu) [inline app/models/exam_template.rb]: See comment about `=>` in previous pull request.

File: app/controllers/exam_templates_controller.rb [hunk 2]
```diff
@@ -20,18 +20,39 @@ def download
...
-    old_template_filename = old_exam_template.filename
-    old_exam_template.replace_with_file(new_uploaded_io.read, assignment_id: assignment.id, filename: old_template_filename)
-    flash_message(:success, t('exam_templates.update.success'))
+
     redirect_to action: 'index'
   end
+
+  def exam_template_params
+    params.require(:exam_template).
+       permit(
```
Comment (david-yz-liu) [inline app/controllers/exam_templates_controller.rb]: Put the `.` on the same line as the method call, not before it.

File: app/views/exam_templates/_form.html.erb [hunk 3]
```diff
@@ -14,6 +14,20 @@
...
+
+    <fieldset>
+      <legend><%= ExamTemplate.human_attribute_name('template_divisions') %></legend>
+      <p><strong><%= t('exam_templates.update.template_divisions') %></strong></p>
+      <p><%= raw t('assignment.required_fields') %> <span class='required_field'>*</span></p>
+      <% @select = Array.new(exam_template.num_pages) { |index| index + 1 }%>
+      <%= f.fields_for :template_divisions do |form| %>
+        <p>
+          <%= form.text_field :label, required: true %>
+          <%= form.select(:start, options_for_select(@select, form.object.start)) %>
```
Comment (david-yz-liu) [inline app/views/exam_templates/_form.html.erb]: `select` is the wrong type of tag to use. It should be a number input.

File: app/views/exam_templates/_form.html.erb [hunk 4]
```diff
@@ -14,6 +14,20 @@
...
       <%= f.label t('exam_templates.update.instruction') %>
       <%= f.file_field :new_template %>
-      <p><%= submit_tag t(:submit)%></p>
     </fieldset>
+
+    <fieldset>
+      <legend><%= ExamTemplate.human_attribute_name('template_divisions') %></legend>
+      <p><strong><%= t('exam_templates.update.template_divisions') %></strong></p>
+      <p><%= raw t('assignment.required_fields') %> <span class='required_field'>*</span></p>
+      <% @select = Array.new(exam_template.num_pages) { |index| index + 1 }%>
```
Comment (david-yz-liu) [inline app/views/exam_templates/_form.html.erb]: This is no longer necessary.

File: app/views/exam_templates/_form.html.erb [hunk 5]
```diff
@@ -14,6 +14,20 @@
...
+      <%= f.fields_for :template_divisions do |form| %>
+        <p>
+          <%= form.text_field :label, required: true %>
+          <%= form.number_field(:start) %>
```
Comment (david-yz-liu) [inline app/views/exam_templates/_form.html.erb]: Good - please pass in the necessary options to ensure that these inputs are >= 1.

**Comment by jenachoi** [comment]
still working on it

**Comment by david-yz-liu** [comment]
Please fix the merge conflicts here.

**Pull request merged**